src/components/Login.jsx

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable jsx-a11y/label-has-for */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import { connect } from 'react-redux';
+import validator from 'validator';
+import PropTypes from 'prop-types';
+import userLogin from '../actions/loginActions';
+import UserForm from '../views/UserForm';
+
+
+/**
+ * @class Login - The Login component
+ * @extends {React.Component}
+ */
+export class Login extends React.Component {
+  /**
+   * @static {propTypes} - validating the prop types supplied to the component
+   * @memberof Login
+   */
+  static propTypes = {
+    userLogin: PropTypes.func,
+    user: PropTypes.object,
+    location: PropTypes.object.isRequired,
+  }
+
+  /**
+   * @static {defaultProps} - providing default values for the props supplied to the component
+   * @function {userLogin} - a default function
+   * @object {user} - a default object 
+   */
+  static defaultProps = {
+    userLogin: f => f,
+    user: {},
+  }
+
+  /**
+   *Creates an instance of Login.
+   * @param {*} props - various props supplied to the component
+   * @memberof Login 
+   */
+  constructor(props) {
```

Comment:
Nice job @thislekan, however, you do not need the constructor method here, just your state will handle what you want to handle.

Reply:
@Andraquin Thank you for your kind feedback.

I'll like to know what the constructor method has against the functionality of this component. What exactly is it that makes the constructor something I need to change?

Reply:
#### Nitpick
yea, I think Andra is right. You can use the ES7 syntax. But you would have to use arrow functions as methods to avoid issues that arises from the 'this' keyword. Great job so far.